db/migrate/20181105061336_cast_key_types_and_values_in_parameters.rb

```diff
@@ -0,0 +1,46 @@
+class CastKeyTypesAndValuesInParameters < ActiveRecord::Migration[5.2]
+  def up
+    User.as_anonymous_admin do
+      Parameter.unscoped.each do |parameter|
+        next if parameter.value.contains_erb?
+        next unless parameter.value.is_a? String
+        override_key_type_and_value(parameter)
+      end
+    end
+  end
+
+  private
+
+  def override_key_type_and_value(param)
+    key_type_name = 'string'
+    value = YAML.load param.value
+    key_type_name = value.is_a?(Hash) ? 'yaml' : find_key_type(value)
+  rescue Psych::SyntaxError
+    value = JSON.load param.value.gsub('=>', ':')
+    key_type_name = find_key_type(value)
+  rescue StandardError, LoadError => e
+    value = param.value
+    say "Failed to cast #{value} for #{param.inspect}"
+    say "Error: #{e.message}"
+  ensure
+    save_param(param, value, key_type_name)
+  end
+
+  def save_param(param, new_value, new_key_type)
+    if Parameter::KEY_TYPES.include?(N_(new_key_type))
```

Comment:
the `N_()` extraction is not needed here, strings are extracted in `Parameter::KEY_TYPES` definition, here we just compare untranslated values